Makefile

```diff
@@ -79,20 +81,46 @@ testhtml: test-internal
 	@rm -f $(test_log) &> /dev/null
 
 .PHONY: test-ci-unit
-test-ci-unit: test-base verify-gen
+test-ci-unit: install-ci-tools test-base verify-gen
 	@echo "+ $@"
 	$(codecov_push) $(coverfile)
 
-.PHONY: install-mockgen
-install-mockgen: install-vendor-dep
+.PHONY: install-ci-tools
+install-ci-tools: install-vendor-dep install-codegen-tools dep-ensure install-mockgen
+	@echo "+ $@"
+	@which gocov > /dev/null || go get github.com/axw/gocov/gocov
+
+.PHONY: install-vendor-dep
+install-vendor-dep:
 	@echo "+ $@"
-	@which goveralls > /dev/null || (go get github.com/golang/mock/gomock && \
-		go install github.com/golang/mock/mockgen)
+	@which dep > /dev/null || (curl https://raw.githubusercontent.com/golang/dep/$(DEP_VERSION)/install.sh | DEP_RELEASE_TAG=$(DEP_VERSION) sh)
```

Comment:
if you decide to use this path for dep, could you include the script in the repo or (ci-scripts). Paranoid about `curl | bash`

Reply:
I'll go with `retool` as mentioned above